libraries/AP_Relay/AP_Relay.cpp

```diff
@@ -195,6 +363,40 @@ bool AP_Relay::arming_checks(size_t buflen, char *buffer) const
     return true;
 }
 
+bool AP_Relay::get(uint8_t instance) const
+{
+    if (instance >= AP_RELAY_NUM_RELAYS) {
+        // invalid instance
+        return false;
+    }
+
+    const int8_t pin = _params[instance].pin.get();
+
+    if (pin < 0) {
+        // invalid pin
+        return false;
+    }
+
+    return (bool)hal.gpio->read(pin);
+}
+
+// see if the relay is enabled
+bool AP_Relay::enabled(uint8_t instance) const 
+{
+    // Must be a valid instance with function relay and pin set
+    return (instance < AP_RELAY_NUM_RELAYS) && (_params[instance].pin != -1) && (_params[instance].function == AP_Relay_Params::FUNCTION::RELAY);
```

Comment:
`ARRAY_SIZE`

... similarly elsewhere.  I like to see the sanity check match the thing we're just about to dereference, *without* going and finding the array  declaration.

Reply:
Have moved to array size throughout.